labs/RivePlayer/src/RivePlayer.cs

```diff
@@ -0,0 +1,255 @@
+                {
+                    IBuffer buffer = await FileIO.ReadBufferAsync(file);
+                    stream = buffer.AsStream();
+                }
+            }
+            if (stream != null && sourceToken == _currentSourceToken)
+            {
+                byte[] data = new byte[stream.Length];
+                stream.Read(data, 0, data.Length);
+                stream.Dispose();  // Don't keep the file open.
+                sceneActionsQueue.Enqueue(() => UpdateScene(SceneUpdates.File, data));
+                // Apply deferred state machine inputs once the scene is fully loaded.
+                foreach (Action stateMachineInput in _deferredSMInputsDuringAsyncSourceLoad!)
+                {
+                    sceneActionsQueue.Enqueue(stateMachineInput);
+                }
+            }
+        }
+        _deferredSMInputsDuringAsyncSourceLoad = null;
+    }
+
+    private void EnqueueStateMachineInput(Action stateMachineInput)
+    {
+        if (_deferredSMInputsDuringAsyncSourceLoad != null)
+        {
+            // A source file is currently loading async. Don't set this input until it completes.
+            _deferredSMInputsDuringAsyncSourceLoad.Add(stateMachineInput);
+        }
+        else
+        {
+            sceneActionsQueue.Enqueue(stateMachineInput);
+        }
+    }
+
+    public void SetBool(string name, bool value)
+    {
+        EnqueueStateMachineInput(() => _scene.SetBool(name, value));
+    }
+
+    public void SetNumber(string name, float value)
+    {
+        EnqueueStateMachineInput(() => _scene.SetNumber(name, value));
+    }
+
+    public void FireTrigger(string name)
+    {
+        EnqueueStateMachineInput(() => _scene.FireTrigger(name));
+    }
+
+    private delegate void PointerHandler(Vec2D pos);
+
+    private void HandlePointerEvent(PointerHandler handler, PointerRoutedEventArgs e)
+    {
+        if (_deferredSMInputsDuringAsyncSourceLoad != null)
+        {
+            // Ignore pointer events while a new scene is loading.
+            return;
+        }
+
+        // Capture the viewSize and pointerPos at the time of the event.
+        var viewSize = this.ActualSize;
+        var pointerPos = e.GetCurrentPoint(this).Position;
+
+        // Forward the pointer event to the render thread.
+        sceneActionsQueue.Enqueue(() =>
+        {
+            Mat2D mat = ComputeAlignment(viewSize.X, viewSize.Y);
+            if (mat.Invert(out var inverse))
+            {
+                Vec2D artboardPos = inverse * new Vec2D((float)pointerPos.X, (float)pointerPos.Y);
+                handler(artboardPos);
+            }
+        });
+    }
+
+    // _scene is used on the render thread exclusively.
+    Scene _scene = new Scene();
+
+    // Source actions originating from other threads must be funneled through this queue.
+    readonly ConcurrentQueue<Action> sceneActionsQueue = new ConcurrentQueue<Action>();
```

Comment:
Nit: use target-typed `new()` when initializing fields directly.
```suggestion
    readonly ConcurrentQueue<Action> sceneActionsQueue = new();
```